### ISSUE TITLE
[TypeScript] Decorator should return ReactN.Component

### ISSUE THREAD

**Issue by CharlesStover** [CharlesStover/reactn#38]
```JavaScript
@reactn
class X extends React.Component {
  render() {
    this.global; // property does not exist (it does)
  }
}
```

This is a TypeScript error, not a runtime error.
Tests pass with `@ts-ignore` on "property does not exist" errors.

**Comment by sumithmano** [comment]
@CharlesStover I would like to give it a try, if you guide me. I looked for the solution, but not getting any lead.

**Comment by CharlesStover** [comment]
I'm not sure how to fix it either. :)

The code is [on this line](https://github.com/CharlesStover/reactn/blob/master/index.d.ts#L92). It should be defining a function that takes in `React.Component<PropType, StateType>` and returns `ReactN.Compoennt<PropType, StateType>` where `ReactN.Component<P, S> extends React.Component<P, S>` with some existing methods and member variables.

I am not good with the exact meanings of `declare class`, `interface`, and `typeof` in TypeScript, but it is currently for some reason or another defining a function that takes `React.Component` as an argument and returns `React.Component` as well (despite it saying that it returns a `GlobalComponent`). In reality, it is returning a `ReactN.Component` and unit tests verify this. There is just something wrong with the definition file. It needs to say that it returns a global Component, and I haven't for the life of me been able to get it to actually say that.

**Comment by CharlesStover** [comment]
As an update, this should be fixed in the 2.0 release, as it is written entirely in TypeScript.
